mettle/src/tlv.c

```diff
@@ -574,18 +568,16 @@ int tlv_dispatcher_process_request(struct tlv_dispatcher *td, struct tlv_packet
 /*
  * This works around garbage on the socket by attempting to read past it.
  */
-bool tlv_have_sync_packet(struct buffer_queue *q, const char *method)
+bool tlv_found_first_packet(struct buffer_queue *q)
 {
 	bool found = false;
-	size_t method_len = strlen(method);
 
-	while (buffer_queue_len(q) >= 62 + method_len) {
+	while (buffer_queue_len(q) >= 571) {
```

Comment:
We were already hardcoding a value there. The idea is that mettle knows approximately what a packet should look like, so it can munch garbage until it finds one. It seems to do a better job the more accurate the parameters are. Unfortunately the actual constant here is tied to whatever metasploit's behavior is.